httpbp/middlewares.go

```diff
@@ -433,7 +434,7 @@ func RecordStatusCode() Middleware {
 // * http_server_requests_total counter with all labels above plus:
 //
 //   - http_response_code: numeric status code as a string, e.g. 200
-func PrometheusServerMetrics(serverSlug string) Middleware {
+func PrometheusServerMetrics() Middleware {
```

Comment:
uh this is a breaking change. we can't cut this as `v0.9.3`.

Reply:
it looks like we accidentally added an arg that's not really used. so in this version we should just use `""` in the default middleware function, and add a TODO to remove this arg from this function in a future, breaking change.

Reply:
we can also rename this arg into `unused` and add some comment in the doc comment of this function to explain that this arg is unused and will be removed in a future version.

Reply:
:(

Reply:
My policy on breaking changes is that we can make them as long as it doesn't break anyone, _especially_ since this is still technically v0.  I am not aware of this API being used outside of baseplate.go, so we should fix it and fix it before it does become used.

If you really want to, we can retract v0.9.2

See the baseplate 2022 compatibility promise:

https://docs.google.com/document/d/1RD5bDhmnQir-gNyEMFWiZjtyc6c8_s51smHkPNlbr4w/edit#heading=h.tsja10upcmb0

Reply:
A breaking change is a breaking change. no one uses it doesn't make it non-breaking.

If someone is already using `v0.9.2` and using this function, breaking it in `v0.9.3` will mean that they need to make a breaking change when they upgrade to `v0.9.3`. retracting `v0.9.2` will help preventing new services from using it (so we probably should still do), but it does not help people already using it.

if no one is using this function from `v0.9.2`, then after `v0.9.3` is cut no one needs to call it any more anyways (because it's already in the default middlewares), and when we make the breaking change in `v0.10.0` they don't need to fix it because they are not using it directly (but via the default middlewares that will also be fixed in `v0.10.0`).